src/modules/chuck-norris/utils/scrape-facts.ts

```diff
@@ -0,0 +1,102 @@
+
+import {writeFileSync} from 'fs';
+const path = require("path"); 
+    
+const Chuck  = require('chucknorris-io'),
+client = new Chuck();
+
+const GetRandomFactForCategory = async( category): Promise<any> =>{
+    var joke = await client.getRandomJoke(category);
+    joke.categories.push(category); // fix bug #2 found in NodeJS Module
+    return joke;
+}
+
+/**
+ * Is the fact on a block list?
+ * @param joke 
+ * @param blockListedFacts 
+ * @returns 
+ */
+function IsBlockListed(joke: any, blockListedFacts) {
+    return blockListedFacts.filter((id) =>{return id === joke.id})
+}
+
+/**
+ * Is the fact already in a list
+ * @param joke 
+ * @param chuckFacts 
+ * @returns 
+ */
+function AlreadyOnList(joke: any, chuckFacts: any[]) {
+    return chuckFacts.filter((element) =>{return element.id === joke.id})
+}
+
+(async () => {
+
+    // Configuration
+    // -------------
+    const excludedCategories = ["explicit"]     // Catgories from which Chuck Norris facts won't be retrieved.  
+    const blockListedFacts = [                  // A list of fact that were identified during testing that may be offensive, and won't be included in this data extract
+        "hyQV9kHDSN61rRIsYwSr4Q",
+        "66Eivn6tSV2IaNTzXEdgFQ",
+        "HNTUHUUDSMObOHUHJul5sw",
+        "-j1T5SaZT3yb1rwfKNDJvQ",
+        "BwMFn3FCQQaCSw0uEKkWLw"
+    ]
+    const FactsPerCateory = 200                // Number of records for each category too download
+    var RetryCount = 10;                        // Number of times to re-attempt an extract for a category.
+
+    // Arguments
+    const destination = process.argv[2];
+    const OutputFile = path.resolve(destination);   
+    
+    console.log("👊 Obtaining Chuck Norris facts")
```

Comment:
Do you mind removing the console.logs or wrap them in a .env file setting for knowing if you are in the dev env. Maybe we could make a generic logger as a ticket to look into that does this for us. So instead you would use something like 
mockedApi.log()

Reply:
@ageddesi - This is the scraping script - this is called by issuing the command `npm chuck` - which then downloads a bunch of chuck norris facts.

Reply:
Ahh ok cool. Thanks @computamike